src/HashMap.mo

```diff
@@ -219,4 +230,18 @@ module {
     h2
   };
 
+  /// The mutable state of a `HashMap`
+  public type S<K,V> = {
+    var table : [var KVs<K, V>];
+    var _count : Nat;
+  };
+
+  /// Creating the state of an empty `HashMap`
+  public func newS<K,V>() : S<K,V>{
```

Comment:
You could turn `S` into a class and avoid some code duplication.

Reply:
> Actually, it could be a public type of the class itself, right?

My understanding is that the type system (still) does not permit type members of classes, only modules, or maybe that changed?